Documentation/op-guide/v2-migration.md

```diff
@@ -0,0 +1,47 @@
+# Migrate applications from using API v2 to API v3
+
+The data store v2 is still accessible from the API v2 after upgrading to etcd3. Thus, it will work as before and require no application changes. With etcd 3, applications use the new grpc API v3 to access the mvcc store, which provides more features and improved performance. The mvcc store and the old store v2 are separate and isolated; writes to the store v2 will not affect the mvcc store and, similarly, writes to the mvcc store will not affect the store v2.
+
+Migrating an application from the API v2 to the API v3 involves two steps: 1) migrate the client library and, 2) migrate the data. If the application can rebuild the data, then migrating the data is unnecessary.
+
+## Migrate client library
+
+API v3 is different from API v2, thus application developers need to use a new client library to send requests to etcd API v3. The documentation of the client v3 is available at https://godoc.org/github.com/coreos/etcd/clientv3. 
+
+There are some notable differences between API v2 and API v3:
+
+- Transaction: In v3, etcd provides capability to do multi-key transaction with conditional flow. Applications need to use transactions instead of Compare-And-Swap.
```

Comment:
Transactions: In v3, etcd provides multi-key conditional transactions. Applications should use transactions in place of Compare-And-Swap operations.